File: Java/P009.java
// Project Euler
// Problem 9: a, b, c, are a Pythagorean triplet. If a + b + c = 100
//            Find the product of abc

public class P009 {
    public static void main(String[] args) {
        System.out.println(compute());
    }

    public static String compute() {
        int a = 1, b = 1, c = 1;

        for (a = 998; a > 1; a--) {
            for (b = 1; b <= (1000-a)/2; b++) {
                c = 1000 - a - b;

                if ((a*a) == ((b*b) + (c*c)))
                    return Integer.toString(a*b*c);
            }
        }
        return Integer.toString(a*b*c);
    }
}

File: Active/TypeScript/src/P009/P009.ts
// Project Euler
// Problem 9: a, b, c, are a Pythagorean triplet. If a + b + c = 100
//            Find the product of abc

export function compute009(sum: number): number {
    for (let a = sum - 2; a > 1; a--) {
        for (let b = 1; b <= (sum - a)/2; b++) {
            let c: number = sum - a - b;

            if ((a * a) === (b * b) + (c * c)) {
                return a * b * c;
            }
        }
    }

    return 0;
}

function run009(): void {
    const sum: number = 1000;
    console.log(compute009(sum).toString());
}

// run009();
